src/modules/sync/hooks/use-sync-dialog-texts.tsx

```diff
@@ -1,46 +1,103 @@
+					),
+				},
+				staging: {
+					title: __( 'Pull from Staging' ),
+					description: __(
+						"Pulling will overwrite your Studio site's selected files and database with a copy from your staging site. Unchecked items will not be changed."
+					),
+				},
+				development: {
+					title: __( 'Pull from Development' ),
+					description: __(
+						"Pulling will overwrite your Studio site's selected files and database with a copy from your development site. Unchecked items will not be changed."
+					),
+				},
+				local: {
+					title: __( 'Pull from remote Local' ),
+					description: __(
+						"Pulling will overwrite your Studio site's selected files and database with a copy from your local site. Unchecked items will not be changed."
+					),
+				},
+			};
 			return {
-				/* translators: %s is the environment name (e.g., "Production", "Staging", "Sandbox") */
-				title: sprintf( __( 'Pull from %s' ), envLabel ),
-				/* translators: %s is the environment type (e.g., "production", "staging", "sandbox") */
-				description: sprintf(
-					__(
-						"Pulling will overwrite your Studio site's selected files and database with a copy from your %s site. Unchecked items will not be changed."
-					),
-					envType
-				),
+				title: textByEnvironment[ siteEnv ].title,
+				description: textByEnvironment[ siteEnv ].description,
 				fromLabel: __( 'Pull' ),
+				toLabel: __( 'To' ),
 				subtitleSelector: __( 'What would you like to pull?' ),
 				envSync: __( 'Read more about <a>environment pull <ArrowIcon /></a>' ),
 				submit: __( 'Pull' ),
 			};
 		} else {
+			const textByEnvironment: TextByEnvironment = {
+				production: {
+					title: __( 'Push to Production' ),
+					description: __(
+						"Pushing will overwrite your production site's selected files and database with content from your local site. Unchecked items will not be changed. The production site will be backed up before any changes are applied."
+					),
+				},
+				staging: {
+					title: __( 'Push to Staging' ),
+					description: __(
+						"Pushing will overwrite your staging site's selected files and database with content from your local site. Unchecked items will not be changed. The staging site will be backed up before any changes are applied."
+					),
+				},
+				development: {
+					title: __( 'Push to Development' ),
+					description: __(
+						"Pushing will overwrite your development site's selected files and database with content from your local site. Unchecked items will not be changed. The development site will be backed up before any changes are applied."
+					),
+				},
+				local: {
+					title: __( 'Push to remote Local' ),
+					description: __(
+						"Pushing will overwrite your local site's selected files and database with content from your local site. Unchecked items will not be changed. The local site will be backed up before any changes are applied."
```

Comment:
Similar as in the equivalent pull text above:

```suggestion
						"Pushing will overwrite your remote local site's selected files and database with content from your local site. Unchecked items will not be changed. The local site will be backed up before any changes are applied."
```

Reply:
I preferred to avoid mentioning "local," so I just used "remote site" and "Studio site" in the description for this case.
https://github.com/Automattic/studio/pull/1648/commits/be8bde22e837d7a547ed85575a3a66c2d228c0a1
https://github.com/Automattic/studio/pull/1648/commits/3462069a817bed611142ad8403fe90f7455d04fa

Reply:
Now reading it again I agree it sounds better with your updated copy. 👍🏼